cmd/cat-main.go

```diff
@@ -62,7 +70,9 @@ EXAMPLES:
 
    3. Concatenate multiple files to one.
       $ {{.HelpName}} part.* > complete.img
-
+   
+   4. Stream a server encrypted object from Amazon S3 cloud storage to standard output.
+      $ {{.HelpName}} --encrypt-key 's3/ferenginar=32byteslongsecretkeymustbegiven1' s3/ferenginar/klingon_opera_aktuh_maylotah.ogg
```

Comment:
@poornas, I wonder what happens if '=' is present in the prefix or in the password. The same problem is what happens if the password contains a space.

I think you can use cli.StringSliceFlag here and probably oblige the user to supply base64 format of the key field, base64 doesn't seem to generate ':' character so the final result can be like that:
`mc cat --encrypt-key 's3/whatever:myplainpassword' 's3/another:mysecondplainkey' --encrypt-b64key 's3/whaterver:CBwbGVhc3VyZS4=' 'myminio/anotherwhatever:dxfwbGVhc3VyZS4' `

What do you think ?

Reply:
@vadmeste, the ui has already been decided to not allow base64 key but to specifically ask for a 32 byte key. Your other points on avoiding '=' or spaces is valid. Will update the pr with more robust parsing

Reply:
@vadmeste , updated the PR , PTAL.

Reply:
> @vadmeste , updated the PR , PTAL.

Cool, I saw your change.

S3 encryption key can support a space in the key, I've just checked. I think we really need to do array style instead of space style in this case. Array as argument is supported in cli package.

Reply:
@vadmeste, since base64 encoding doesn't seem to use "," we can switch to that as field separator. using string slices seems to entail specifying --encrypt-key alias/bucket=key1 --encrypt-key alias/bucket/prefix=key2 which would become cumbersome for the user. What do you think?

Reply:
> @vadmeste, since base64 encoding doesn't seem to use "," we can switch to that as field separator. using string slices seems to entail specifying --encrypt-key alias/bucket=key1 --encrypt-key alias/bucket/prefix=key2 which would become cumbersome for the user. What do you think?

In this case, I think both ',' and space would work.

But if we will be using base64, then we shouldn't use = as a field separator since = exists in base64 result. So I guess ':' can be useful in this case, like:
`--encrypt-key 'myminio/:plainpasswordetc s3/:bXlsb25nc2VjcmV0a2V5Cg=='`

One last thing, what about if we support both plain and base64 password at the same time ?

`--encrypt-key "myminio/:plain-32-bytes-long-secret-key"` and `--encrypt-key "myminio/:base64,bXlsb25nc2VjcmV0a2V5Cg==`

Otherwise your PR is all good for me.

Reply:
@vadmeste, the current regex will recognize the first = and not parse the = in the key, so that is safe.Updated PR  to use , as delimiter for k-v pairs. Think that should take care of your concerns.

As far as accepting plain and base64 keys - think it will provide less intuitive user experience than just sticking to one input option.